src/cmap/auth/mongodb_aws.ts

```diff
@@ -108,8 +108,8 @@ export class MongoDBAWS extends AuthProvider {
           return;
         }
 
-        // TODO(NODE-4990)
         if (!ByteUtils.equals(serverNonce.subarray(0, nonce.byteLength), nonce)) {
+          // The serverNonce's leading 32 bytes must equal the client nonce's 32 bytes
```

Comment:
should this comment be above this block? and also, I thought you were going to at least mention the spec in case someone else is wondering where this comes from

Reply:
I usually prefer to put comments inside the `if`, just a habit, trying to make the code read like "if X is true, do Y" so if I'm following my own practice I should write something more like: `// throw because the server's nonce must begin with...` kinda like how mocha test the "it" is supposed to imply "the function under test should..."

will add a link to the spec

Reply:
https://github.com/mongodb/specifications/blob/875446db44aade414011731840831f38a6c668df/source/auth/auth.rst#id11

I used the latest commit hash incase a future version changes the header link